src/SwiftRuntimeLibrary/src/SwiftCore.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<swifttypedatabase version="1.0">
+    <entities>
+        <entity managedNameSpace="System" managedTypeName="Byte" entityType="Scalar" blittable="true" size="1" stride="1">
+            <typedeclaration kind="struct" name="Swift.UInt8" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="SByte" entityType="Scalar" blittable="true" size="1" stride="1">
+            <typedeclaration kind="struct" name="Swift.Int8" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int16" entityType="Scalar" blittable="true" size="2" stride="2">
+            <typedeclaration kind="struct" name="Swift.Int16" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt16" entityType="Scalar" blittable="true" size="2" stride="2">
+            <typedeclaration kind="struct" name="Swift.UInt16" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int32" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.Int32" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt32" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.UInt32" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int64" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Int64" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt64" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.UInt64" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Single" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.Float" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Double" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Double" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="IntPtr" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Int" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UIntPtr" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.UInt" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="bool" entityType="Scalar" blittable="false" size="4" stride="4">
```

Comment:
Is this for .NET, Swift, or the marshaled type? Both .NET and Swift have `bool` as 1 byte - `MemoryLayout<Bool>.size`. The built in marshaller defaults to the `BOOL` size, which is 4. However, if we are documenting that here then I don't understand what this file is documenting. Is it the .NET, Swift or interop types? What are we doing about [`DisableDefaultMarshalling`](https://learn.microsoft.com/dotnet/api/system.runtime.compilerservices.disableruntimemarshallingattribute), since in that case `bool` would indeed be 1.

Reply:
This file is meant for mapping swift types to .NET types. In a quick look-over of the code, size, stride and blittable are artifacts and can go away entirely.
In the dim dark ages of BTfS, I tried to create two classes of struct: blittable (and fully representable in C# with no wrappers) and non-blittable. This was an endless source of bugs and I removed it.